### ISSUE TITLE
IDE0010 False positive case var

### ISSUE THREAD

**Issue by philipborg** [dotnet/roslyn#71281]
**Version Used**: 

dotnet SDK 8.0.100, using built in analyzers

**Steps to Reproduce**:

1. Enable IDE0044.
2. Write code:
```csharp
switch (1)
{
	case var namedDefault:
		// Log or throw namedDefault code, though content isn't important.
		break;
}
```
3. Get warning from IDE0010 because of lack of default case. A `case var` acts as a default case however, thus making it a false positive.

**Diagnostic Id**:

`IDE0010`

**Expected Behavior**:

No analyzer trigger as there is effectively a default branch.

**Actual Behavior**:

Analyzer triggers on the sample code.

**Comment by philipborg** [comment]
Impressive response time @CyrusNajmabadi!

**Comment by CyrusNajmabadi** [comment]
Thanks :-)
